torch/_dynamo/variables/builder.py

```diff
@@ -755,7 +772,10 @@ def index_source(key):
                 new_symint == 1,
             )
         else:
-            result = UserDefinedObjectVariable(
+            var_cls = UserDefinedObjectVariable
+            if is_torch_function_user_object(value):
+                var_cls = TorchFunctionObjectVariable
```

Comment:
This TorchFunctionObjectVariable doesn't make much sense to me. Torch function is triggered solely by the presence of the `__torch_function__` method. It seems to me that the handling should just be stuck in UserDefinedObjectVariable directly (where UDOV should know how to accurately report if it has `__torch_function__` or not.)